src/rust-crypto/backup.ts

```diff
@@ -495,8 +497,18 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
      */
     private keysCountInBatch(batch: RustSdkCryptoJs.KeysBackupRequest): number {
         const parsedBody: KeyBackup = JSON.parse(batch.body);
+        return this.getTotalKeyCount(parsedBody);
+    }
+
+    /**
+     * This method calculates the total number of keys present in a key backup
+     * @param keyBackup - The key backup to count the keys from.
+     *
+     * @returns The total number of keys in the backup.
+     */
+    private getTotalKeyCount(keyBackup: KeyBackup): number {
         let count = 0;
-        for (const { sessions } of Object.values(parsedBody.rooms)) {
+        for (const { sessions } of Object.values(keyBackup.rooms)) {
             count += Object.keys(sessions).length;
         }
         return count;
```

Comment:
given that this is independent of `RustBackupManager`, make it a standalone utility function?